lib/matplotlib/legend.py

```diff
@@ -444,28 +444,15 @@ def __init__(self, parent, handles, labels,
                 loc = 'upper right'
         if isinstance(loc, str):
             if loc not in self.codes:
```

Comment:
Would this now better be a `try except KeyError: raise ValueError`?

Reply:
I think that just clutters the traceback:
```ipynb
In [2]: plt.figure().legend(loc='ca')
```
with the `try`, adds one more traceback frame, which is not really any clearer to anyone:
```pytb
KeyError                                  Traceback (most recent call last)
~/code/matplotlib/lib/matplotlib/legend.py in __init__(self, parent, handles, labels, loc, numpoints, markerscale, markerfirst, scatterpoints, scatteryoffsets, prop, fontsize, borderpad, labelspacing, handlelength, handleheight, handletextpad, borderaxespad, columnspacing, ncol, mode, fancybox, shadow, title, title_fontsize, framealpha, edgecolor, facecolor, bbox_to_anchor, bbox_transform, frameon, handler_map)
    446             try:
--> 447                 loc = self.codes[loc]
    448             except KeyError:

KeyError: 'ca'

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
<ipython-input-2-c48871c74701> in <module>()
----> 1 plt.figure().legend(loc='ca')

~/code/matplotlib/lib/matplotlib/figure.py in legend(self, *args, **kwargs)
   1827             # extra_args = extra_args[1:]
   1828             pass
-> 1829         l = mlegend.Legend(self, handles, labels, *extra_args, **kwargs)
   1830         self.legends.append(l)
   1831         l._remove_method = self.legends.remove

~/code/matplotlib/lib/matplotlib/legend.py in __init__(self, parent, handles, labels, loc, numpoints, markerscale, markerfirst, scatterpoints, scatteryoffsets, prop, fontsize, borderpad, labelspacing, handlelength, handleheight, handletextpad, borderaxespad, columnspacing, ncol, mode, fancybox, shadow, title, title_fontsize, framealpha, edgecolor, facecolor, bbox_to_anchor, bbox_transform, frameon, handler_map)
    449                 raise ValueError(
    450                     "Unrecognized location {!r}. Valid locations are\n\t{}\n"
--> 451                     .format(loc, '\n\t'.join(self.codes)))
    452         if not self.isaxes and loc == 0:
    453             raise ValueError(

ValueError: Unrecognized location 'ca'. Valid locations are
	best
	upper right
	upper left
	lower left
	lower right
	right
	center left
	center right
	lower center
	upper center
	center
```